src/backend/api/meals.js

```diff
@@ -4,8 +4,94 @@ const knex = require("../database");
 
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
+    if (request.query.limit && request.query.maxPrice){
+      const limit=request.query.limit;
+      const maxPrice=request.query.maxPrice;
+      const limitedMeals=await knex("meals").where("price","<=",maxPrice).limit(limit);
+
+    response.json(limitedMeals)
+    }
```

Comment:
You are also checking for the limit and the maxPrice a few lines down, so this seems unnecessary!

Reply:
"Only specific number of meals with a specific max price( query parameters and you should make sure that you can combine them - see example limit+maxPrice)" this is one of the tasks which is the combination of two other parameters, isnt it? :)

Reply:
Yes that is correct, however this combination limit+maxPrice is just an example, all possible combinations should work (eg title+maxPrice) and it'd be really ugly if you tried to do this for every single one of them!
Watch Orhan's video which I linked, and it'll become clearer :)